src/screens/SUE/SueReportScreen.tsx

```diff
@@ -27,12 +27,47 @@ import {
 } from '../../components';
 import { colors, device, normalize, texts } from '../../config';
 import { addToStore, readFromStore } from '../../helpers';
-import { useKeyboardHeight, useStaticContent } from '../../hooks';
+import { useKeyboardHeight } from '../../hooks';
 import { QUERY_TYPES, getQuery } from '../../queries';
 import { postRequests } from '../../queries/SUE';
 
 export const SUE_REPORT_VALUES = 'sueReportValues';
 
+type TRequiredFields = {
+  [key: string]: {
+    [key: string]: boolean;
+  };
+};
+
+const sueProgressWithRequiredInputs = (
+  progress: TProgress[],
+  fields: TRequiredFields
+): TProgress[] => {
+  const requiredInputs: { [key: string]: boolean } = {};
+
+  for (const section in fields) {
+    for (const field in fields[section]) {
+      requiredInputs[field] = fields[section][field];
+    }
+  }
+
+  return progress.map((item) => {
+    if (!item.requiredInputs) {
+      item.requiredInputs = [];
+    }
+
+    item.requiredInputs = item.requiredInputs.filter((key) => requiredInputs[key] !== false);
```

Comment:
what about reducing this to:

```
item.requiredInputs = (item.requiredInputs || []).filter((key) => requiredInputs[key] !== false);
```

Reply:
Yes, it makes sense. Then we don't need the if check above.